actionview/lib/action_view/template.rb

```diff
@@ -324,8 +325,11 @@ def handle_render_error(view, e) #:nodoc:
       end
 
       def locals_code #:nodoc:
+        locals = @locals.to_set - Module::DELEGATION_RESERVED_METHOD_NAMES
+        locals = locals.select { |key| /\A[a-z_][a-zA-Z_0-9]*\z/.match?(key) }
```

Comment:
Did not find a reference for regexp. This regex is incorrect, was looking for something that would handle unicode but haven't found anything useful yet (e.g. something that matches a single emoji as a valid identifier.) Open to suggestions!